### DISCUSSION TITLE
Fastpi issues with GCP CloudRun vs GCP Compute

### DISCUSSION THREAD

**Discussion by sidbenac** [fastapi/fastapi#9950] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
FROM tiangolo/uvicorn-gunicorn:python3.11 
ENV PYTHONUNBUFFERED 1
ENV PORT 8000
ENV GUNICORN_NUM_WORKERS 4

    gunicorn main:app -k uvicorn.workers.UvicornWorker -w $GUNICORN_NUM_WORKERS -b 0.0.0.0:8000
```

### Description

I have issues running fastapi server in cloud run. let me explain : 
I have setup 2 servers. one on gcp cloud run. the same as a VPC in compute. here is the call to run gunicorn : 

    gunicorn main:app -k uvicorn.workers.UvicornWorker -w $GUNICORN_NUM_WORKERS -b 0.0.0.0:8000 

I send a stress test using hey command to send 10000 call.

when I send the stress test in cloud run the server stops and I start having workers timeout signal 6

the same stress test on the VPC the server responds perfectly and all is working fine.

What am I doing wrong. I use Async/Await and it seems that something is wrong with cloud run handling traffic

has anyone came accross the same issue ?

### Operating System

Linux

### Operating System Details

Cloud run docker through load balancer
VPC direct machine through IP

### FastAPI Version

latest

### Python Version

3.11

**Comment by antonio-cinnamon** [comment]
@sidbenac Were you able to figure out what was happening? I'm having the same issue with cloud run timing out and memory exceeding.
